webapp/app/js/compareperf.js

```diff
@@ -118,9 +115,20 @@ perf.controller('CompareResultsCtrl', [
           cmap.detailsLink = detailsLink;
           cmap.name = platform;
           cmap.hideMinorChanges = $scope.hideMinorChanges;
+          if (Object.keys($scope.compareResults).indexOf(testName) < 0)
+            $scope.compareResults[testName] = [];
           $scope.compareResults[testName].push(cmap);
         });
       });
+
+      // Remove the tests with no data, report them as well; not needed for subtests
+      var noresults = [];
+      _.difference($scope.testList, Object.keys($scope.compareResults)).forEach(function(name) {
+        noresults.push(' ' + name.replace(' summary', ''));
+      });
```

Comment:
You could do this all in one shot with map, instead of iterating with foreach

$scope.testNoResults = _.difference($scope.testList, Object.keys($scope.compareResults)).map(function(name) { return ' ' + name.replace(' summary', '') });